xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/exporter/exporter.js

```diff
@@ -152,60 +154,188 @@ require([
+
+            var node = treeReference.get_node(nodeDom);
+            var subNodeId = node.data.id;
+            var subPageReference;
+
+            // if the node does not have children (easy case)
+            if (treeReference.is_leaf(node)) {
+
+              // pagination node is actually a leaf node
+              if (node.data.type === "pagination") {
+                isPaginationExisting = true;
+                isPaginationChecked = treeReference.is_checked(node);
+
+              // standard document node
+              } else {
+                subPageReference = XWiki.Model.serialize(XWiki.Model.resolve(subNodeId, XWiki.EntityType.DOCUMENT));
+
+                // the page ref in the right list
+                if (treeReference.is_checked(node)) {
+                  includedPages.push(subPageReference);
+                } else {
+                  excludedPages.push(subPageReference);
+                }
               }
-              // Special behaviour for the XAR export
-              if (url.params.format == 'xar') {
-                // Also add the WebPreferences
-                var webPreferencesReference = new XWiki.EntityReference('WebPreferences', XWiki.EntityType.DOCUMENT, 
-                  pageReference.parent);
-                url.params.pages.push(XWiki.Model.serialize(webPreferencesReference));
+
+            // if the node has children
+            } else {
+              subPageReference = XWiki.Model.resolve(subNodeId, XWiki.EntityType.DOCUMENT);
+              var subPageJoker = XWiki.Model.serialize(new XWiki.EntityReference('%', XWiki.EntityType.DOCUMENT, subPageReference.parent));
+
+              // if it's not loaded, we'll manage the entire space
+              if (!treeReference.is_loaded(node)) {
+
+                // put the space in the right list
+                if (treeReference.is_checked(node)) {
+                  includedPages.push(subPageJoker);
+                } else {
+                  excludedPages.push(subPageJoker);
+                }
+              } else {
+                // in that case the page will be managed within its own space, so we need to exclude it
+                // in the current context to avoid selecting it in the request
+                // e.g. I'm in Foo.% I selected Foo.Bar but only some of its children:
+                // I need Foo.Bar.% to be excluded in the request with Foo.%
+                // then another part of the request will select Foo.Bar.X, X being the selected children.
+                excludedPages.push(subPageJoker);
+                nodeWithChildren.push(node);
               }
             }
+          });
+
+          // if we don't have a pagination node, or we have one and it's checked:
+          // we manage the export with exporting the space and excluding stuff that needs to be
+          if (!isPaginationExisting || (isPaginationExisting && isPaginationChecked)) {
+            includedPages = [ pageJoker ];
```

Comment:
What's the point of setting ``includedPages`` here? You don't use it after this.

Reply:
indeed. Old stuff I guess. Will remove it.